### PR TITLE
SHT30 driver integration

### PR DESCRIPTION
**Pull request by marvin-g-personal**
## Changes

- updated Cerberus to work with the new sht 

## Notes

- kept major code the same

## To Do

- should be done

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No merge conflicts
- [ ] All checks passing
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #160)

### CODE REVIEW

**Comment by marvin-g-personal** [comment]
is the build failing because embedded base hadn't been updated with the new sht?

File: Core/Src/monitor.c [hunk 1]
```diff
@@ -260,7 +260,7 @@ void vTempMonitor(void *pv_params)
 		/* Take measurement */
 		uint16_t temp = 0;
 		uint16_t humidity = 0;
-		if (read_temp_sensor(mpu, &temp, &humidity)) {
+		if (read_temp_sensor(mpu, &temp, &humidity) != 0) {
```
Comment (dyldonahue) [inline Core/Src/monitor.c:263]: previous convention is better, please do not change style conventions unless its part of the ticket

File: Core/Src/mpu.c [hunk 2]
```diff
@@ -35,10 +35,13 @@ mpu_t *init_mpu(I2C_HandleTypeDef *hi2c, ADC_HandleTypeDef *pedals_adc,
 	mpu->watchdog_gpio = watchdog_gpio;
 
 	/* Initialize the Onboard Temperature Sensor */
-	mpu->temp_sensor = malloc(sizeof(sht30_t));
-	assert(mpu->temp_sensor);
-	mpu->temp_sensor->i2c_handle = hi2c;
-	assert(!sht30_init(mpu->temp_sensor)); /* This is always connected */
+	// updated the sht30 initialization
+	mpu->temp_sensor.i2c_handle = hi2c;
```
Comment (dyldonahue) [inline Core/Src/mpu.c]: based on ur (correct) change to the driver, there is no i2c handle in this struct anymore, so this needs to be removed

File: Core/Src/mpu.c [hunk 3]
```diff
@@ -35,10 +35,13 @@ mpu_t *init_mpu(I2C_HandleTypeDef *hi2c, ADC_HandleTypeDef *pedals_adc,
...
+	mpu->temp_sensor.write = (Write_ptr)HAL_I2C_Master_Transmit;
+	mpu->temp_sensor.read = (Read_Ptr)HAL_I2C_Master_Receive;
+	mpu->temp_sensor.mem_read = (Mem_Read_Ptr)HAL_I2C_Mem_Read;
```
Comment (dyldonahue) [inline Core/Src/mpu.c]: this wont work. The signature for the HAL functions is different than the function pointers you created. Youll need to create wrappers for them. Take a look at Caios work with the lsm driver

File: Core/Src/mpu.c [hunk 4]
```diff
@@ -150,9 +154,8 @@ int8_t read_accel(mpu_t *mpu, uint16_t accel[3])
 		return mut_stat;
 
 	HAL_StatusTypeDef hal_stat = lsm6dso_read_accel(mpu->imu);
-	if (hal_stat)
+	if (hal_stat != HAL_OK)
```
Comment (dyldonahue) [inline Core/Src/mpu.c]: again, please dont change this convention. (PS - it sort of becomes totally obvious that ur just putting this stuff through AI when arbitrary changes like this - in a completely different spot than any relevant changes u made - appear) Our code is open sourced so its whatever, but u can get in huge trouble with any employers for doing that with proprietary code. If im way off base my fault, but based on some changes here and in the embedded pr i assume i am not

**Review by dyldonahue — CHANGES_REQUESTED** [review]

**Comment by dyldonahue** [comment]
in general, yes this build check is going to fail bc it refrecnes embedded base without ur changes. But there are still mistakes that would cause it to fail with urd atm. You should always try building locally with ur embedded base commit - I cant imagine this builds locally bc of that i2c handle member that no longer exists

**Comment by marvin-g-personal** [comment]
I created a new branch that has the current changes caio made from develop since I made my original one in main; this will help prevent merge conflicts, I'll name the new PR the same thing; mb.

**Pull request closed without merging**